Mage.Sets/src/mage/cards/h/HauntingApparition.java

```diff
@@ -0,0 +1,88 @@
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.IntPlusDynamicValue;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.ChooseOpponentEffect;
+import mage.abilities.effects.common.continuous.SetBasePowerSourceEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreatureCard;
+import mage.filter.predicate.mageobject.ColorPredicate;
+import mage.game.Game;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ * @author Cguy7777
+ */
+public final class HauntingApparition extends CardImpl {
+
+    public HauntingApparition(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{U}{B}");
+
+        this.subtype.add(SubType.SPIRIT);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(2);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // As Haunting Apparition enters the battlefield, choose an opponent.
+        this.addAbility(new AsEntersBattlefieldAbility(new ChooseOpponentEffect(Outcome.Detriment)));
+
+        // Haunting Apparition's power is equal to 1 plus the number of green creature cards in the chosen player's graveyard.
+        Effect effect = new SetBasePowerSourceEffect(
+                new IntPlusDynamicValue(1, new greenCreatureCardsInChosenOpponentsGraveyardCount()));
+        effect.setText("{this}'s power is equal to 1 plus the number of green creature cards in the chosen player's graveyard");
+        this.addAbility(new SimpleStaticAbility(Zone.ALL, effect));
+    }
+
+    private HauntingApparition(final HauntingApparition card) {
+        super(card);
+    }
+
+    @Override
+    public HauntingApparition copy() {
+        return new HauntingApparition(this);
+    }
+}
+
+class greenCreatureCardsInChosenOpponentsGraveyardCount implements DynamicValue {
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        if (sourceAbility != null) {
+            UUID playerId = (UUID) game.getState().getValue(sourceAbility.getSourceId() + ChooseOpponentEffect.VALUE_KEY);
+            Player chosenOpponent = game.getPlayer(playerId);
+            if (chosenOpponent != null) {
+                FilterCreatureCard filter = new FilterCreatureCard();
+                filter.add(new ColorPredicate(ObjectColor.GREEN));
+
+                return chosenOpponent.getGraveyard().count(filter, game);
+            }
+        }
+        return 0;
+    }
+
+    @Override
+    public greenCreatureCardsInChosenOpponentsGraveyardCount copy() {
+        return new greenCreatureCardsInChosenOpponentsGraveyardCount();
+    }
+
+    @Override
+    public String getMessage() {
```

Comment:
you might also want to override `toString()` to return `1`, `X`, or whatever makes the text work with this class... or maybe not needed here, not sure.

Reply:
I don't *think* it matters the way things are right now. The text has to be set manually, because it doesn't generate correctly with `SetBasePowerSourceEffect()`. It comes out as "Haunting Apparition’s power is equal to **the number of** 1 plus the number of green...", like cards affected by #11542. (I actually discovered that while implementing this card.)